usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlock.java

```diff
@@ -0,0 +1,264 @@
+                final SortedMap<Integer, StopTime> stopTimeCollectionForCurrentTrip =
+                        dataRepo.getStopTimeByTripId(currentTrip.getTripId());
+
+                final StopTime currentTripFirstStopTimeInSequence =
+                        stopTimeCollectionForCurrentTrip.get(stopTimeCollectionForCurrentTrip.firstKey());
+                final StopTime currentTripLastStopTimeInSequence =
+                        stopTimeCollectionForCurrentTrip.get(stopTimeCollectionForCurrentTrip.lastKey());
+                final Integer currentTripFirstTime = currentTripFirstStopTimeInSequence.getArrivalTime();
+                final Integer currentTripLastTime = currentTripLastStopTimeInSequence.getDepartureTime();
+
+                visitedTripIdCollection.add(currentTripId);
+
+                if (currentTripFirstTime != null && currentTripLastTime != null) {
+                    if (currentTripFirstTime <= currentTripLastTime) {
+                        tripCollectionFromSameBlockId.forEach(unvisitedTrip -> {
+                            final String unvisitedTripId = unvisitedTrip.getTripId();
+                            if (!visitedTripIdCollection.contains(unvisitedTripId)) {
+                                // if trips have the same service_id
+                                // | routeId | tripId | serviceId | blockId | first stop time | last stop time |
+                                // |---------|--------|-----------|---------|-----------------|----------------|
+                                // | 0       | 2      | a         | 5       | 07:00           | 10:00          |
+                                // | 0       | 5      | a         | 5       | 08:45           | 11:20          |
+                                // | 0       | 8      | a         | 5       | 11:40           | 13:50          |
+
+                                // fetch related StopTime for unvisitedTrip
+                                final SortedMap<Integer, StopTime> stopTimesForUnvisitedTrip =
+                                        dataRepo.getStopTimeByTripId(unvisitedTripId);
+
+                                final StopTime unvisitedTripFirstStopTimeInSequence =
+                                        stopTimesForUnvisitedTrip.get(stopTimesForUnvisitedTrip.firstKey());
+                                final StopTime unvisitedTripLastStopTimeInSequence =
+                                        stopTimesForUnvisitedTrip.get(stopTimesForUnvisitedTrip.lastKey());
+
+                                final Integer unvisitedTripFirstTime =
+                                        unvisitedTripFirstStopTimeInSequence.getArrivalTime();
+                                final Integer unvisitedTripLastTime =
+                                        unvisitedTripLastStopTimeInSequence.getDepartureTime();
+
+                                if (unvisitedTrip.getServiceId().equals(currentTrip.getServiceId())) {
```

Comment:
Is service ID guaranteed not to be `null` here?

Reply:
Indeed, the GTFS specification required this field to be provided, and we have a null check in `TripBuilder` `.build` method.